CMakeLists.txt

```diff
@@ -21,8 +21,11 @@ if (NOT DEFINED ONNX_TARGET_NAME)
   set(ONNX_TARGET_NAME onnx)
 endif()
 
-option(protobuf_BUILD_TESTS "" OFF)
-add_subdirectory_if_no_target(${PROJECT_SOURCE_DIR}/third_party/protobuf/cmake libprotobuf)
+if(NOT ${USE_SYSTEM_PROTOBUF})
```

Comment:
`if (NOT ${xxx})` will be interpreted as `if (NOT )` if xxx is not defined or empty,so

```cmake
# tmp.cmake

if (NOT xxx)
    message(STATUS "hhh")
else()
    message(STATUS "ddd")
endif()

if (NOT ${xxx})
    message(STATUS "hhh")
else()
    message(STATUS "ddd")
endif()
```

```bash
$ cmake -P tmp.cmake
-- hhh
-- ddd
```

Reply:
and please add a `ONNX_OPT_USE_SYSTEM_PROTOBUF` option

Reply:
you are right